xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -1455,10 +1454,36 @@ void CGUIMediaWindow::OnInitWindow()
 
   m_rootDir.SetAllowThreads(true);
 
-  if (m_iSelectedItem > -1)
-    m_viewControl.SetSelectedItem(m_iSelectedItem);
-
   CGUIWindow::OnInitWindow();
+
+  m_viewControl.SetSelectedItem(GetMediaSelectedItem(m_vecItems->GetPath()));
+}
+
+int CGUIMediaWindow::GetMediaSelectedItem(const std::string& strDirectory)
+{
+  std::map<std::string, int>::iterator it;
+  it = std::find_if(m_mediaSelectedItems.begin(), m_mediaSelectedItems.end(), [&strDirectory](const std::pair<std::string, int>& it)
```

Comment:
with maps you can directly use map::find to check if the key (first) exists.
http://en.cppreference.com/w/cpp/container/map/find